p2p/upgrade.go

```diff
@@ -182,7 +182,7 @@ func Upgrade(h host.Host, opts ...Opt) (*Host, error) {
 	if fh.cfg.EnableRoutingDiscovery {
 		dopts = append(dopts, discovery.EnableRoutingDiscovery())
 	}
-	if !fh.cfg.RoutingDiscoveryNoAdvertise {
+	if !fh.cfg.RoutingDiscoveryAdvertise {
```

Comment:
Could it be that this should be

```suggestion
	if fh.cfg.RoutingDiscoveryAdvertise {
```

?

Reply:
Good catch! Broke this part in a last minute update... Fixed.